File: Sorting/SelectionSort.java
// Java Code to implement Selection Sorting Technique

import java.util.*;
public class SelectionSort{
    public static void main(String args[])
    {
        Scanner sc=new Scanner(System.in);
        System.out.println("Input:\n");
        System.out.println("Enter the size of an array : \n");
        int size=sc.nextInt();
        sc.nextLine();
        int arr[]=new int[size];
        System.out.println("Enter the elements of an array :\n");
        for(int index = 0; index < size; index++) {
             arr[index]=sc.nextInt();
        }
        //Algorithms for Selection Sorting
        for(int k=0;k<size-1;k++)
        {
            int indx=k;
            for(int j=k+1;j<size;j++)
            {
                if(arr[indx]>arr[j])
                {
                    indx=j;
                }
            }
            //Swapping of values when found minimum 
            int temp=arr[indx];
            arr[indx]=arr[k];
            arr[k]=temp;
        }
        //Displaying Output
        System.out.println("Output:\n");
        for(int i=0;i<size;i++)
        {
            System.out.print(arr[i]+" ");
        }
        
    }
}

/*
Input:

Enter the size of an array : 5 

Enter the elements of an array :
64 22 12 25 11

Output:

11 12 22 25 64 

Time Complexity: O(n^2)
Space Complexity: O(1)

*/

File: Sorting/Selection Sort.cpp
#include < iostream >
using namespace std;
 
void selectionSort(int arr[]) {
  for (int i = 0; i < 4; i++) {
    int min = i;
 
    for (int j = i + 1; j < 5; j++) {
      if (arr[j] < arr[min]) {
        min = j;
      }
    }
    if (min != i) {
      int temp = arr[min];
      arr[min] = arr[i];
      arr[i] = temp;
    }
  }
}
 
int main() {
 
  int myarr[5];
  cout << "Enter 5 integers in random order: " << endl;
  for (int i = 0; i < 5; i++) {
    cin >> myarr[i];
  }
 
  cout << "UNSORTED ARRAY: " << endl;
  for (int i = 0; i < 5; i++) {
    cout << myarr[i] << "  ";
  }
  cout << endl;
 
  selectionSort(myarr); // sorting actually happening
 
  cout << "SORTED ARRAY: " << endl;
  for (int i = 0; i < 5; i++) {
    cout << myarr[i] << "  ";
  }
  return 0;
}
